### PR TITLE
Padding iphone x

### LINKED ISSUE #782: Trays need extra padding on iPhone X
# 🐛 Bug Report 

Trays on iPhone X or other iPhones with no home button are partially obscured by the home indicator at the bottom of the screen.

## 🤔 Expected Behavior

There should be some extra padding at the bottom of the tray to account for the home indicator.

## 😯 Current Behavior

The home indicator sometimes overlaps or is too close to the contents of the tray, making it hard to access.

## 💁 Possible Solution

We could possibly add some padding using the safe area insets defined in CSS. Not tried this yet though. https://webkit.org/blog/7929/designing-websites-for-iphone-x/

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3.0.2
| Browser          | Safari
| Operating System | iOS

## 🧢 Your Company/Team

RSP

### PR DESCRIPTION
**Pull request by ronderksen**
Closes #782

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Open the picker examples on an iPhone X or newer (without a physical home button) in portrait mode and try to tap the last item.
The solution still isn't perfect, but it seems to at least clear the home bar at the bottom of the screen

## 🧢 Your Project:

None

### CODE REVIEW

File: packages/@adobe/spectrum-css-temp/components/tray/index.css [hunk 1]
```diff
@@ -66,6 +66,10 @@
   transition: opacity var(--spectrum-dialog-exit-animation-duration) cubic-bezier(0.5, 0, 1, 1) var(--spectrum-dialog-exit-animation-delay),
     transform var(--spectrum-dialog-exit-animation-duration) cubic-bezier(0.5, 0, 1, 1) var(--spectrum-dialog-exit-animation-delay);
 
+  &:last-child {
+    padding-bottom: calc(env(safe-area-inset-bottom) + 8px);
```
Comment (snowystinger) [inline packages/@adobe/spectrum-css-temp/components/tray/index.css:70]: while a really neat function that i just learned about (thank you!)
it seems it relies on a specific meta tag? https://developer.mozilla.org/en-US/docs/Web/CSS/env
> To tell the browser to use the whole available space on the screen, and so enabling us to use the env() variables, we need to add a new viewport meta value:
is this not the case?

Comment (devongovett) [reply packages/@adobe/spectrum-css-temp/components/tray/index.css:70]: hmm well we don't have `viewport-fit=cover` in the docs site, just `width=device-width, initial-scale=1` and we still have this issue...

Comment (ronderksen) [reply packages/@adobe/spectrum-css-temp/components/tray/index.css:70]: You don't need the meta tag part. I've checked and I definitely see a difference in padding with and without the env() part.

**Comment by devongovett** [comment]
Thanks @ronderksen! I've removed the extra 8px of padding in addition to the safe area inset because that would affect non-iPhones and also when the Safari toolbar is visible. Also, the tray is always the last child of the tray wrapper element, so always adding the padding is fine.

I assume the extra 8px was to try to avoid tapping near the bottom showing the toolbar and hiding the tray due to scrolling. I'm going to open a separate PR to prevent that.

**Comment by devongovett** [comment]
#1023

**Pull request merged**